### PR TITLE
feat: custom errors in Marketplace.sol

### PR DESCRIPTION
**Pull request by robrobbins**

### CODE REVIEW

**Review by sourabhmarathe — APPROVED** [review]

File: build/marketplace/MarketPlace.sol [hunk 1]
```diff
@@ -67,11 +72,20 @@ contract MarketPlace {
     string memory n,
     string memory s
   ) external authorized(admin) unpaused() returns (bool) {
-    require(swivel != address(0), 'swivel contract address not set');
+    // TODO remove on confirm
+    // require(swivel != address(0), 'swivel contract address not set');
+    if (swivel == address(0)) {
+      revert Exception(21, 0, 0, address(0), address(0));
```
Comment (JTraversa) [inline build/marketplace/MarketPlace.sol:78]: Might be able to get away with a generic "address not set" if there are other occurrences in Swivel.sol itself?

File: build/marketplace/MarketPlace.sol [hunk 2]
```diff
@@ -96,15 +110,28 @@ contract MarketPlace {
...
+    // require(block.timestamp >= m, 'maturity not reached');
+    if (block.timestamp < m) {
+      revert Exception(24, block.timestamp, m, address(0), address(0));
+    }
 
     // set the base maturity cToken exchange rate at maturity to the current cToken exchange rate
     uint256 exchangeRate = Compounding.exchangeRate(p, market.cTokenAddr);
     markets[p][u][m].maturityRate = exchangeRate;
 
     // TODO remove the require here? why?
```
Comment (JTraversa) [inline build/marketplace/MarketPlace.sol]: I think just remove it imo.

Comment (robrobbins) [reply build/marketplace/MarketPlace.sol]: done

File: build/marketplace/MarketPlace.sol (hunk 1 above)
Comment (robrobbins) [reply build/marketplace/MarketPlace.sol:78]: there's not any others, it's specific to the need of having swivel's address set.

plus, the list is @alexanderwende approved so ...

**Pull request merged**